scanpy/tests/test_plotting.py

```diff
@@ -1685,3 +1685,10 @@ def test_string_mask(tmp_path, check_same_image):
     plt.close()
 
     check_same_image(p1, p2, tol=1)
+
+
+def test_violin_scale_warning(monkeypatch):
+    adata = pbmc3k_processed()
+    monkeypatch.setattr(sc.pl.StackedViolin, "DEFAULT_SCALE", "count", raising=False)
+    with pytest.warns(FutureWarning, match="Don’t set DEFAULT_SCALE"):
+        sc.pl.StackedViolin(adata, adata.var_names[:3], groupby="louvain")
```

Comment:
This is just spitballing, but do you think it's worth adding a test so that these functions don't raise __any__ warnings (of a certain kind, perhaps)?  Just a loose idea, but would it be easier to permit certain warnings than to make a test every time we come across a warning that should be squashed?

Reply:
For example, at the moment, I get 

```python
In [5]: sc.pl.rank_genes_groups_violin(adata, groups='0', n_genes=8, density_norm='count')
/Users/ilangold/Projects/Theis/scanpy/scanpy/plotting/_tools/__init__.py:1287: FutureWarning:

Setting a gradient palette using color= is deprecated and will be removed in v0.14.0. Set `palette='dark:black'` for the same effect.

  _ax = sns.stripplot(
/Users/ilangold/Projects/Theis/scanpy/scanpy/plotting/_tools/__init__.py:1303: UserWarning: set_ticklabels() should only be used with a fixed number of ticks, i.e. after set_ticks() or using a FixedLocator.
  _ax.set_xticklabels(new_gene_names, rotation="vertical")
```

Reply:
Got it.